### DISCUSSION TITLE
accessing the calendar through groovy scripts

### DISCUSSION THREAD

**Discussion by MajidAmpaewia** [freeplane/freeplane#932] [Help]
Dear community,
I am a complete newcomer and would like to briefly thank everyone who has made freeplane what it is today!
I have only been programming only a little in groovy for a short time and ask for your understanding if my knowledge is not yet great and I may have to ask something that is easily apparent to others. I am interested in accessing the calendar through groovy to set timers and calendar entries and potentially manipulate them dynamically with my own script. I would be very happy to receive a response!
Sincerely,
majid

**Comment by EdoFro** [comment]
look at the Freeplane's API:
1. click Menu: `Help->Freeplane API…`
2. browser opens showing the Freeplane's API documentation
3. search for "reminder"
![image](https://user-images.githubusercontent.com/35700575/211628504-b4d79da6-20d2-4135-857b-66bb9b47600b.png)

There you will see all the properties and methods associated with reminders

### Some examples
1. this script creates a periodic reminder. To make the reminder fire every second week:
```groovy
node.reminder.createOrReplace(new Date() + 1, "WEEK", 2)
```

2. to show the data of the selected node's reminder:
```groovy
def r = node.reminder

def t = "Selected node's reminder:"
    t += '\nRemind at: ' + r.remindAt
    t += '\nPeriod: ' +r.period + ' ' + r. periodUnit

ui.showMessage(t.toString(),1)
```
hope it helps

edo

**Reply by MajidAmpaewia** [reply]
Perfect, that's exactly what I meant. Thank you very much!
